src/azul/terra.py

```diff
@@ -500,6 +501,11 @@ def run_sql(self, query: str) -> BigQueryRows:
                         sleep(delay)
                     else:
                         raise e
+                except InternalServerError as e:
+                    log.warning('Server error from BigQuery job during attempt %i/%i. '
+                                'Retrying in %is.',
+                                attempt + 1, len(delays) + 1, delay, exc_info=e)
+                    sleep(delay)
```

Comment:
Too much duplicated code for my taste. Consider a different branching structure for the if 

But most importantly, the condition on `delay is not None` seems to be missing for this branch.